All of the test failures look like this:
```
FAILED tests/_test_msui/test_wms_control.py::Test_HSecWMSControlWidget::test_no_server - TypeError: WMSControlWidgetSetup.query_server() missing 1 required positional argument: 'qtbot'
```
What it means is that all tests that use query_server have to pass along the qtbot fixture that they receive. I.e. the calls to query_server have to be adapted as well, when its signature changes.

Also, there are some minor linting issues from flake8, just click on "Details" of that check to see what they are.

Other than that it is looking good already. Just a minor nitpick: I would like `qtbot` to be the first argument to `query_server`, instead of the last. Just to be consistent with other places where this kind of thing is done, e.g. here: https://github.com/Open-MSS/MSS/blob/d4e8d6b39ea8ca83c1aed405df0d49c8d19ddb4f/tests/_test_msui/test_mscolab.py#L229

--- a/tests/_test_msui/test_linearview.py
+++ b/tests/_test_msui/test_linearview.py
@@ -106,7 +106,7 @@ class Test_LinearViewWMS:
         self.window.hide()
         shutil.rmtree(self.tempdir)
 
-    def query_server(self, url, qtbot):
+    def query_server(self, qtbot, url):
         QtTest.QTest.keyClicks(self.wms_control.multilayers.cbWMS_URL, url)
         with qtbot.wait_signal(self.wms_control.cpdlg.canceled):
             QtTest.QTest.mouseClick(self.wms_control.multilayers.btGetCapabilities, QtCore.Qt.LeftButton)
@@ -115,6 +115,6 @@ class Test_LinearViewWMS:
         """
         assert that a getmap call to a WMS server displays an image
         """
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
         with qtbot.wait_signal(self.wms_control.image_displayed):
             QtTest.QTest.mouseClick(self.wms_control.btGetMap, QtCore.Qt.LeftButton)

--- a/tests/_test_msui/test_sideview.py
+++ b/tests/_test_msui/test_sideview.py
@@ -146,7 +146,7 @@ class Test_SideViewWMS:
         self.window.hide()
         shutil.rmtree(self.tempdir)
 
-    def query_server(self, url, qtbot):
+    def query_server(self, qtbot, url):
         QtTest.QTest.keyClicks(self.wms_control.multilayers.cbWMS_URL, url)
         with qtbot.wait_signal(self.wms_control.cpdlg.canceled):
             QtTest.QTest.mouseClick(self.wms_control.multilayers.btGetCapabilities, QtCore.Qt.LeftButton)
@@ -155,7 +155,7 @@ class Test_SideViewWMS:
         """
         assert that a getmap call to a WMS server displays an image
         """
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
         with qtbot.wait_signal(self.wms_control.image_displayed):
             QtTest.QTest.mouseClick(self.wms_control.btGetMap, QtCore.Qt.LeftButton)
         assert self.window.getView().plotter.image is not None

--- a/tests/_test_msui/test_topview.py
+++ b/tests/_test_msui/test_topview.py
@@ -221,7 +221,7 @@ class Test_TopViewWMS:
         self.window.hide()
         shutil.rmtree(self.tempdir)
 
-    def query_server(self, url, qtbot):
+    def query_server(self, qtbot, url):
         QtTest.QTest.keyClicks(self.wms_control.multilayers.cbWMS_URL, url)
         with qtbot.wait_signal(self.wms_control.cpdlg.canceled):
             QtTest.QTest.mouseClick(self.wms_control.multilayers.btGetCapabilities, QtCore.Qt.LeftButton)
@@ -230,7 +230,7 @@ class Test_TopViewWMS:
         """
         assert that a getmap call to a WMS server displays an image
         """
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
         with qtbot.wait_signal(self.wms_control.image_displayed):
             QtTest.QTest.mouseClick(self.wms_control.btGetMap, QtCore.Qt.LeftButton)
         assert self.window.getView().map.image is not None

--- a/tests/_test_msui/test_wms_control.py
+++ b/tests/_test_msui/test_wms_control.py
@@ -106,40 +106,45 @@ class Test_HSecWMSControlWidget(WMSControlWidgetSetup):
         """
         assert that a message box informs about server troubles
         """
+        mock_url = f"{self.scheme}://{self.host}:{self.port-1}"
         with mock.patch("PyQt5.QtWidgets.QMessageBox.critical") as mock_critical:
-            self.query_server(f"{self.scheme}://{self.host}:{self.port-1}")
+            self.query_server(mock_url)
             mock_critical.assert_called_once()
 
     def test_no_schema(self):
         """
         assert that a message box informs about server troubles
         """
+        mock_url = f"{self.host}:{self.port}"
         with mock.patch("PyQt5.QtWidgets.QMessageBox.critical") as mock_critical:
-            self.query_server(f"{self.host}:{self.port}")
+            self.query_server(mock_url)
             mock_critical.assert_called_once()
 
     def test_invalid_schema(self):
         """
         assert that a message box informs about server troubles
         """
+        mock_url = f"hppd://{self.host}:{self.port}"
         with mock.patch("PyQt5.QtWidgets.QMessageBox.critical") as mock_critical:
-            self.query_server(f"hppd://{self.host}:{self.port}")
+            self.query_server(mock_url)
             mock_critical.assert_called_once()
 
     def test_invalid_url(self):
         """
         assert that a message box informs about server troubles
         """
+        mock_url = f"{self.scheme}://???{self.host}:{self.port}"
         with mock.patch("PyQt5.QtWidgets.QMessageBox.critical") as mock_critical:
-            self.query_server(f"{self.scheme}://???{self.host}:{self.port}")
+            self.query_server(mock_url)
             mock_critical.assert_called_once()
 
     def test_connection_error(self):
         """
         assert that a message box informs about server troubles
         """
+        mock_url = f"{self.scheme}://.....{self.host}:{self.port}"
         with mock.patch("PyQt5.QtWidgets.QMessageBox.critical") as mock_critical:
-            self.query_server(f"{self.scheme}://.....{self.host}:{self.port}")
+            self.query_server(mock_url)
             mock_critical.assert_called_once()
 
     @pytest.mark.skip("Breaks other tests in this class because of a lingering message box, for some reason")

--- a/tests/_test_msui/test_wms_control.py
+++ b/tests/_test_msui/test_wms_control.py
@@ -87,12 +87,13 @@ class WMSControlWidgetSetup:
         self.window.hide()
         shutil.rmtree(self.tempdir)
 
-    def query_server(self, url, qtbot):
+    def query_server(self, qtbot, url):
         while len(self.window.multilayers.cbWMS_URL.currentText()) > 0:
             QtTest.QTest.keyClick(self.window.multilayers.cbWMS_URL, QtCore.Qt.Key_Backspace)
         QtTest.QTest.keyClicks(self.window.multilayers.cbWMS_URL, url)
         with qtbot.wait_signal(self.window.cpdlg.canceled):
             QtTest.QTest.mouseClick(self.window.multilayers.btGetCapabilities, QtCore.Qt.LeftButton)
+
 
 class Test_HSecWMSControlWidget(WMSControlWidgetSetup):
     @pytest.fixture(autouse=True)
@@ -164,7 +165,7 @@ class Test_HSecWMSControlWidget(WMSControlWidgetSetup):
         """
         assert that an aborted getmap call does not change the displayed image
         """
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
         QtTest.QTest.keyClick(self.window.pdlg, QtCore.Qt.Key_Enter)
         with qtbot.wait_signal(self.window.image_displayed):
             QtTest.QTest.mouseClick(self.window.btGetMap, QtCore.Qt.LeftButton)
@@ -177,7 +178,7 @@ class Test_HSecWMSControlWidget(WMSControlWidgetSetup):
         """
         assert that a getmap call to a WMS server displays an image
         """
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
 
         with qtbot.wait_signal(self.window.image_displayed):
             QtTest.QTest.mouseClick(self.window.btGetMap, QtCore.Qt.LeftButton)
@@ -190,7 +191,7 @@ class Test_HSecWMSControlWidget(WMSControlWidgetSetup):
         """
         assert that a getmap call to a WMS server displays an image
         """
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
 
         with qtbot.wait_signal(self.window.image_displayed):
             QtTest.QTest.mouseClick(self.window.btGetMap, QtCore.Qt.LeftButton)
@@ -212,7 +213,7 @@ class Test_HSecWMSControlWidget(WMSControlWidgetSetup):
         """
         assert that changing between servers still allows image retrieval
         """
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
 
         with mock.patch("PyQt5.QtWidgets.QMessageBox.critical") as qm_critical:
             with qtbot.wait_signal(self.window.cpdlg.canceled):
@@ -240,7 +241,7 @@ class Test_HSecWMSControlWidget(WMSControlWidgetSetup):
         """
         assert that multilayers get created, handled and drawn properly
         """
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
         server = self.window.multilayers.listLayers.findItems(f"{self.url}/",
                                                               QtCore.Qt.MatchFixedString)[0]
         self.window.cbAutoUpdate.setCheckState(False)
@@ -322,7 +323,7 @@ class Test_HSecWMSControlWidget(WMSControlWidgetSetup):
         """
         assert that singlelayer mode behaves as expected
         """
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
         server = self.window.multilayers.listLayers.findItems(f"{self.url}/",
                                                               QtCore.Qt.MatchFixedString)[0]
         self.window.cbAutoUpdate.setCheckState(False)
@@ -384,7 +385,7 @@ class Test_HSecWMSControlWidget(WMSControlWidgetSetup):
 
     @mock.patch("mslib.msui.wms_control.WMSMapFetcher.moveToThread")
     def test_server_no_thread(self, mockthread, qtbot):
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
         server = self.window.multilayers.listLayers.findItems(f"{self.url}/",
                                                               QtCore.Qt.MatchFixedString)[0]
         self.window.cbAutoUpdate.setCheckState(False)
@@ -417,7 +418,7 @@ class Test_VSecWMSControlWidget(WMSControlWidgetSetup):
         """
         assert that a getmap call to a WMS server displays an image
         """
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
         with qtbot.wait_signal(self.window.image_displayed):
             QtTest.QTest.mouseClick(self.window.btGetMap, QtCore.Qt.LeftButton)
 
@@ -429,11 +430,12 @@ class Test_VSecWMSControlWidget(WMSControlWidgetSetup):
         """
         assert that drawing a layer through code doesn't fail for vsec
         """
-        self.query_server(self.url)
+        self.query_server(self.url, qtbot)
         server = self.window.multilayers.listLayers.findItems(f"{self.url}/",
                                                               QtCore.Qt.MatchFixedString)[0]
         with qtbot.wait_signal(self.window.image_displayed):
             server.child(0).draw()
+
 
 class TestWMSControlWidgetSetupSimple:
     xml = """<?xml version="1.0" encoding="UTF-8" standalone="no"?>

--- a/tests/_test_msui/test_wms_control.py
+++ b/tests/_test_msui/test_wms_control.py
@@ -102,13 +102,13 @@ class Test_HSecWMSControlWidget(WMSControlWidgetSetup):
         yield
         self._teardown()
 
-    def test_no_server(self):
+    def test_no_server(self, qtbot):
         """
         assert that a message box informs about server troubles
         """
         mock_url = f"{self.scheme}://{self.host}:{self.port-1}"
         with mock.patch("PyQt5.QtWidgets.QMessageBox.critical") as mock_critical:
-            self.query_server(mock_url)
+            self.query_server(qtbot, mock_url)
             mock_critical.assert_called_once()
 
     def test_no_schema(self):

